libs/platforms/azure/hypershiftcli/hypershiftcli.py

```diff
@@ -225,9 +228,34 @@ def delete_cluster(self, platform, cluster_name):
             self.logging.error(err)
             self.logging.error(f"Failed to write metadata_destroy.json file located at {cluster_info['path']}")
         if self.es is not None:
-            cluster_info["timestamp"] = datetime.datetime.utcnow().isoformat()
             self.es.index_metadata(cluster_info)
 
+    def wait_for_controlplane_ready(self, cluster_name, wait_time):
+        myenv = os.environ.copy()
+        myenv["KUBECONFIG"] = self.environment['mc_kubeconfig']
+        starting_time = datetime.datetime.utcnow().timestamp()
+        while datetime.datetime.utcnow().timestamp() < starting_time + wait_time * 60:
+            if self.utils.force_terminate:
+                self.logging.error(f"Exiting install times capturing on {cluster_name} cluster after capturing Ctrl-C")
+                return 0
+            self.logging.info(f"Getting cluster information for cluster {cluster_name} on {self.environment['mgmt_cluster_name']}")
+            cluster_status_code, cluster_status_out, cluster_status_err = self.utils.subprocess_exec("oc get hostedcluster -n clusters " + cluster_name + " -o json", extra_params={"env": myenv, "universal_newlines": True})
+            current_time = int(datetime.datetime.utcnow().timestamp())
+            try:
+                cluster_status = json.loads(cluster_status_out).get("status", {}).get("conditions", [])
+            except Exception as err:
+                self.logging.error(f"Cannot load command result for cluster {cluster_name}. Waiting 1 seconds for next check...")
+                self.logging.error(err)
+                time.sleep(1)
+                continue
+            if any(item["message"] == "The hosted control plane is available" and item["status"] == "True" for item in cluster_status):
```

Comment:
@morenod This status also reflects kube-apiserver Availability ? or just the `hostedcontrolplane` custom resource status.

Reply:
verified